src/webgpu/api/validation/queue/copyImageBitmapToTexture.spec.ts

```diff
@@ -0,0 +1,480 @@
+      size: { width: 1, height: 1, depth: 1 },
+      format: 'bgra8unorm',
+      usage: GPUTextureUsage.COPY_DST,
+    });
+
+    t.runTest(
+      { imageBitmap },
+      { texture: isErrDstTexture ? t.getErrorTexture() : correctTexture },
+      { width: 1, height: 1, depth: 1 },
+      !isErrDstTexture
+    );
+  });
+
+g.test('destination_texture_usage')
+  .params(poptions('usage', kTextureUsages))
+  .fn(async t => {
+    const { usage } = t.params;
+    const imageBitmap = await createImageBitmap(t.getImageData(1, 1));
+    const dstTexture = t.device.createTexture({
+      size: { width: 1, height: 1, depth: 1 },
+      format: 'rgba8unorm',
+      usage,
+    });
+
+    t.runTest(
+      { imageBitmap },
+      { texture: dstTexture },
+      { width: 1, height: 1, depth: 1 },
+      !!(usage & GPUTextureUsage.COPY_DST)
+    );
+  });
+
+g.test('destination_texture_sample_count')
+  .params(poptions('sampleCount', [1, 4]))
+  .fn(async t => {
+    const { sampleCount } = t.params;
+    const imageBitmap = await createImageBitmap(t.getImageData(1, 1));
+    const dstTexture = t.device.createTexture({
+      size: { width: 1, height: 1, depth: 1 },
+      sampleCount,
+      format: 'bgra8unorm',
+      usage: GPUTextureUsage.COPY_DST,
+    });
+
+    t.runTest(
+      { imageBitmap },
+      { texture: dstTexture },
+      { width: 1, height: 1, depth: 1 },
+      sampleCount === 1
+    );
+  });
+
+g.test('destination_texture_mipLevel')
+  .params(poptions('mipLevel', [0, kDefaultMipLevelCount - 1, kDefaultMipLevelCount]))
+  .fn(async t => {
+    const { mipLevel } = t.params;
+    const imageBitmap = await createImageBitmap(t.getImageData(1, 1));
+    const dstTexture = t.device.createTexture({
+      size: { width: kDefaultWidth, height: kDefaultHeight, depth: kDefaultDepth },
+      mipLevelCount: kDefaultMipLevelCount,
+      format: 'bgra8unorm',
+      usage: GPUTextureUsage.COPY_DST,
+    });
+
+    t.runTest(
+      { imageBitmap },
+      { texture: dstTexture, mipLevel },
+      { width: 1, height: 1, depth: 1 },
+      mipLevel < kDefaultMipLevelCount
+    );
+  });
+
+g.test('desitnation_texture_format')
+  .params(poptions('format', kAllTextureFormats))
+  .fn(async t => {
+    const { format } = t.params;
+    const imageBitmap = await createImageBitmap(t.getImageData(1, 1));
+
+    // createTexture with all possible texture format may have validation error when using
+    // compressed texture format.
+    t.device.pushErrorScope('validation');
+    const dstTexture = t.device.createTexture({
+      size: { width: 1, height: 1, depth: 1 },
+      format,
+      usage: GPUTextureUsage.COPY_DST,
+    });
+    t.device.popErrorScope();
+
+    const success = kValidTextureFormatsForCopyIB2T.includes(format);
+
+    t.runTest(
+      { imageBitmap },
+      { texture: dstTexture },
+      { width: 1, height: 1, depth: 1 },
+      success,
+      success ? '' : 'TypeError'
+    );
+  });
+
+g.test('copy_view_origin')
```

Comment:
This and `copy_size_OOB ` seem to test the kinda same thing, but it isn't clear what the separation is.

Instead can we have a test for source OOB and a test for destination OOB? The src OOB test would parametrize on copySize and srcOriginValue while the dst OOB test would parametrize on copySize, dstOriginValue and mipLevel.